packages/react-dom/src/__tests__/ReactBrowserEventEmitter-test.js

```diff
@@ -10,151 +10,133 @@
-    }
-
-    class ChildWrapper extends React.PureComponent {
+    class Child extends React.PureComponent {
       render() {
-        return <Child {...this.props} />;
+        return <div {...this.props} />;
       }
     }
 
-    class CommonComponent extends React.Component {
+    ParentComponent = class Parent extends React.Component {
       render() {
         return (
-          <div ref={c => (GRANDPARENT = c)} {...GRANDPARENT_PROPS}>
-            <div ref={c => (PARENT = c)} {...PARENT_PROPS}>
-              <ChildWrapper {...CHILD_PROPS} />
-            </div>
+          <div {...this.props.parentProps}>
+            <Child {...this.props.childProps} />
           </div>
         );
       }
-    }
-
-    function renderTree() {
-      instance = ReactTestUtils.renderIntoDocument(<CommonComponent />);
-    }
-    renderTree();
-
-    putListener = function(node, eventName, listener) {
-      switch (node) {
-        case CHILD:
-          CHILD_PROPS[eventName] = listener;
-          break;
-        case PARENT:
-          PARENT_PROPS[eventName] = listener;
-          break;
-        case GRANDPARENT:
-          GRANDPARENT_PROPS[eventName] = listener;
-          break;
-      }
-      renderTree();
     };
 
-    deleteAllListeners = function(node) {
-      switch (node) {
-        case CHILD:
-          CHILD_PROPS = {};
-          break;
-        case PARENT:
-          PARENT_PROPS = {};
-          break;
-        case GRANDPARENT:
-          GRANDPARENT_PROPS = {};
-          break;
-      }
-      renderTree();
+    /**
+     * alway initial the new instance for child component to bind click listener
+     * @param {}
+     * @return {Component Instance} initial instance
+     */
+    createInitialInstance = () => {
+      const initialInstance = ReactDOM.render(
+        <ParentComponent parentProps={{}} childProps={{onClick: LISTENER}} />,
+        container,
+      );
+      const child = getChildInstance(initialInstance);
+      expect(child.props.onClick).toBe(LISTENER);
+
+      return initialInstance;
     };
 
-    getListener = function(node, eventName) {
-      const trees = ReactTestUtils.findAllInRenderedTree(instance, () => true);
-      return trees[3].props[eventName];
+    /**
+     * update and rerender the same instance
+     * @param {object} parentProps - props for parent component to update
+     * @param {object} childProps - props for child component to update
+     */
+    updateInstance = ({parentProps, childProps}) => {
+      return ReactDOM.render(
+        <ParentComponent parentProps={parentProps} childProps={childProps} />,
+        container,
+      );
     };
 
-    idCallOrder = [];
-  });
+    /**
+     * helper to find the specified instance
+     * @param {Component Instance} inst - main instance
+     * @return {Component Instance} Child instance
+     */
+    getChildInstance = inst => {
+      return ReactTestUtils.findRenderedComponentWithType(inst, Child);
```

Comment:
Can we use refs and avoid the need for this method?

Reply:
oh ! I missing refs can use with Component Class before I read the document again, I remove the get* method and only use refs